src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -304,6 +313,17 @@ void Instance::HandleScanNetworks(HandlerContext & ctx, const Commands::ScanNetw
         if (!nullableSSID.IsNull())
         {
             ssid = nullableSSID.Value();
+            if (ssid.empty())
+            {
+                // Normalize the zero value to null ByteSpan.
+                // Spec 7.17.1. Empty string is an equivalent of null.
+                ssid = ByteSpan();
+            }
+        }
+        if (!(ssid.size() <= DeviceLayer::Internal::kMaxWiFiSSIDLength))
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
+            return;
         }
         mAsyncCommandHandle = CommandHandler::Handle(&ctx.mCommandHandler);
         mpDriver.Get<WiFiDriver *>()->ScanNetworks(ssid, this);
```

Comment:
I still think that we should just have two different ScanNetworks APIs: one taking a (nonempty) SSID, and one not taking an SSID at all, with the latter representing "scan everything", on the driver.

In particular, this change on its own is leaving some backends broken.

Reply:
Don't quite agree, for all platforms, the only difference between directed scanning and non-directed scanning is the  argument passed to scan network.

- mbed os: no directed scanning, result are filtered if it is a directional scanning
- ESP32: `wifi_scan_config_t.ssid`
- P6: scan_filter.mode and `scan_filter.param.SSID`
- EFR32: the first argument of `wfx_start_scan`
- Linux (wpa_supplicant) one field of the argument of the Scan command

On some platforms, even the branch can be omitted

Reply:
>  the only difference between directed scanning and non-directed scanning is the argument passed to scan network.

That's true, but if you look at the actual code, the platforms end up havint quite different code on those two branches, and most importantly doing it this way will make it much harder for a platform to mess this up.

Platforms that really want to have a single codepath can internally do whatever they want to represent one SSID vs all.

> mbed os: no directed scanning, result are filtered if it is a directional scanning

Yes, and it would just common up in their OnScanNetwork callback anyway.

> ESP32: wifi_scan_config_t.ssid

The actual code is:
```
    if (ssid.data())
    {
        wifi_scan_config_t scan_config = { 0 };
        memset(WiFiSSIDStr, 0, sizeof(WiFiSSIDStr));
        memcpy(WiFiSSIDStr, ssid.data(), ssid.size());
        scan_config.ssid = WiFiSSIDStr;
        err              = esp_wifi_scan_start(&scan_config, false);
    }
    else
    {
        err = esp_wifi_scan_start(NULL, false);
    }
```

> P6: scan_filter.mode and scan_filter.param.SSID

Right, totally different scan_filter setup on the two codepaths.

And so on.

But good point about the fact that normalizing to "empty and null both end up with default-constructed ByteSpan" at least does give us consistent behavior on all the backends....